service/history/workflow/update/registry.go

```diff
@@ -361,12 +362,16 @@ func (r *registry) Len() int {
 	return len(r.updates)
 }
 
-// remover is called when Update gets into terminal state (completed or rejected).
+// remover is called when an Update gets into a terminal state (completed or rejected).
 func (r *registry) remover(id string) updateOpt {
 	return withCompletionCallback(
 		func() {
+			// A rejected update is *not* counted as a completed update
+			// as that would negatively impact the registry's rate limit.
+			if updateWasAccepted := r.updates[id].acceptedEventID != common.EmptyEventID; updateWasAccepted {
```

Comment:
The zero value here would be a nil pointer -- did you decide that we don't need to handle the case of `id` not being in the `updates` map at this point?

Reply:
Good that you comment on that. I had a nil check there at first; but then it seemed unnecessary to me. But you never know. I can the nil check back.